docs/KernelFeatures/UsingSecureVault.md

```diff
@@ -0,0 +1,108 @@
+
+There are three files that are needed by the SecureVault:
+
+1. Secure-vault.yaml: 
+    Configurations that are needed to configure the SecureVault is given in this file
+2. Master-keys.yaml: 
+    The default SecureVault implementation is based on the Java Key Store (JKS). Passwords that are needed to access the JKS and Keys are specified in this file. 
+3. Secrets.properties: 
+    This file contains the alias with plain text or encrypted password.
+    Example:
+    
+        UserManager.AdminUser.Password=plainText ABC@123
+        UserManager.AdminUser.Password=cipherText SnBSWKjtZZOo0UsmOpPRhP6ZMNYTb80+BZHRDC/kxNT9ExcTswAbFjb/aip2KgQNaVuIT27UtrBaIv77Mb5sNPGiwyPrfajLNhSOlke2p8YmMkegx/mG2ytJhJa5j9iMGtCsbMt+SAf85v6kGIiH0gZA20qDZ9jnveT7/Ifz7v0\=
+
+The SecureVatult reads the aliases and passwords given in the secrets.properties file and return the resolved (decrypted) password when it is queried wish a valid alias of a password.
+
+The SecureVault implementation has two major sub-components in it, namely Master Key Reader and Secret Repository. The SecureVault implementation allows to plugin custom implementations for both of these:
+
+1. Secret Repository
+   The default implementation of SectetRepository is based on the passwords and aliases given in the Secrets.properties file and the default JKS packed with the carbon product.
+2. Master Key Reader
+   The default implementation of MasterKeyReader gets a list of required passwords from the SecretRepository and provides the values for those passwords by reading system properties, environment variables and master-keys.yaml file.
+
+## How To Use SecureVault
+SecureVault is by default enabled. It reads the aliases and passwords given in the secrets.properties file. The secrets.properties file may contain both plain text and encrypted passwords. We have a separate tool called "ciphertool" (shipped with Carbon /bin) to encrypt the secrets.properties file. Once the tool is run, it will encrypt all the plaintext password in the secrets.properties file.
+
+CipherTool also depends on the configurations given in the secure-vault.yaml. Hence making necessary changes in the "secure-vault.yaml" file before running the CipherTool is mandatory. Once configured, running the "ciphertool" is simple as running the "ciphertool.sh" on Linux and Mac and running the "ciphertool.bat" on Windows machines.
+
+## How To Implement Custom Master Key Reader
+MasterKeyReader (org.wso2.carbon.kernel.securevault.MasterKeyReader) is the interface which all the MasterKeyReader implementations should derive from and should be registered as an OSGi service. SecureVault gets all the MasterKeyReader implementations and binds itself only with the matching MasterKeyReader which is specified in the "secure-vault.yaml" file.
```

Comment:
All the MasterKeyReader implementations should derive from the MasterKeyReader (...) interface. This interface should be registered as an OSGi service. SecureVault.......only with the matching MasterKeyReader, which is specified in the ......